### ISSUE TITLE
encoding/binary: can't compare NativeEndian with LittleEndian/BigEndian

### ISSUE THREAD

**Issue by dolmen** [golang/go#63611]
TL;DR: `binary.NativeEndian == binary.BigEndian` doesn't work.

Follow-up on #57237.

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.21.3 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes.

### What did you do?

https://go.dev/play/p/PIdcmG89Rp-
```go
package main

import (
	"encoding/binary"
	"fmt"
)

func main() {
	fmt.Println(binary.NativeEndian == binary.BigEndian)
}

```

### What did you expect to see?

I expect this program to compile and to allow me to simply determine if the platform is big endian or little endian as a boolean value.

As the author of package [`github.com/dolmen-go/endian`](https://pkg.go.dev/github.com/dolmen-go/endian) which provided [`Native`](https://pkg.go.dev/github.com/dolmen-go/endian#Native) long before and [has this feature](https://github.com/dolmen-go/endian/blob/master/amd64_test.go#L13), I miss that.

### What did you see instead?

```
./prog.go:9:37: invalid operation: binary.NativeEndian == binary.BigEndian (mismatched types binary.nativeEndian and binary.bigEndian)
```

**Comment by bcmills** [comment]
Programs generally shouldn't need to care which endianness is native, especially if they are already using `encoding/binary`. But for the rare ones that do, how about [`golang.org/x/sys/cpu.IsBigEndian`](https://pkg.go.dev/golang.org/x/sys/cpu#pkg-constants)?

**Comment by rittneje** [comment]
From what I recall, this is an artifact of the fact that `binary.LittleEndian` and `binary.BigEndian` are not comparable with each other. So if for example `binary.NativeEndian` were an alias of `binary.LittleEndian` on a little endian system, then `binary.NativeEndian == binary.LittleEndian` would return true on little endian and fail to compile on big endian. The only way around that would have been to make `binary.NativeEndian` an interface, preventing some compile-time optimizations.

I'll note your library has the compilation issue I mentioned, so I don't think it can be used in the way you intended. https://go.dev/play/p/NuQV5ijf5PG

**Comment by cagedmantis** [comment]
cc [@griesemer](https://github.com/griesemer)
